lib/iris/experimental/ugrid.py

```diff
@@ -932,8 +991,75 @@ def __ne__(self, other):
         return NotImplemented
 
     def __repr__(self):
-        # TBD
-        args = []
+        def to_coord_and_axis(members):
+            def axis(member):
+                return member.split("_")[1]
+
+            result = [
+                f"({coord!s}, {axis(member)!r})"
+                for member, coord in members._asdict().items()
+                if coord is not None
+            ]
+            result = f"[{', '.join(result)}]" if result else None
+            return result
+
+        node_coords_and_axes = to_coord_and_axis(self.node_coords)
+        connectivities = [
+            str(connectivity)
+            for connectivity in self.all_connectivities
+            if connectivity is not None
+        ]
+
+        if len(connectivities) == 1:
+            connectivities = connectivities[0]
+        else:
+            connectivities = f"[{', '.join(connectivities)}]"
+
+        # positional arguments
+        args = [
+            f"topology_dimension={self.topology_dimension!r}",
+            f"node_coords_and_axes={node_coords_and_axes}",
+            f"connectivities={connectivities}",
+        ]
+
+        # optional argument
+        edge_coords_and_axes = to_coord_and_axis(self.edge_coords)
+        if edge_coords_and_axes:
+            args.append(f"edge_coords_and_axes={edge_coords_and_axes}")
+
+        # optional argument
+        if self.topology_dimension > 1:
+            face_coords_and_axes = to_coord_and_axis(self.face_coords)
+            if face_coords_and_axes:
+                args.append(f"face_coords_and_axes={face_coords_and_axes}")
+
+        def kwargs_filter(k, v):
+            result = False
+            if k != "topology_dimension":
+                if not (
+                    self.topology_dimension == 1 and k == "face_dimension"
+                ):
+                    if v is not None:
+                        result = True
+                        if (
+                            not isinstance(v, str)
+                            and isinstance(v, Iterable)
+                            and not v
+                        ):
+                            result = False
+                        elif k == "units" and v == "unknown":
+                            result = False
```

Comment:
If the `units` are `unknown`, then don't print them... this is the `units` equivalent to being `None`